src/main/java/Duke.java

```diff
@@ -1,10 +1,148 @@
+        getUserInput();
+
+        //Print closing message
+        printFarewell();
+    }
+
+    public static void loopEcho() {
+        Scanner sc = new Scanner(System.in);
+        String userInput = sc.nextLine();
+        while (!userInput.equals("bye")) {
+            printDivider();
+            System.out.printf("\t%s%n", userInput);
+            printDivider();
+            userInput = sc.nextLine();
+        }
+    }
+
+    public static void getUserInput() {
+        Task[] taskList = new Task[MAX_TASK_LENGTH];
+        int listIndex = 0;
+        Scanner sc = new Scanner(System.in);
+        String userInput = sc.nextLine();
+        while (!userInput.trim().toLowerCase().equals("bye")) {
+            if (userInput.trim().toLowerCase().equals("list")) {
+                displayList(taskList);
+            } else if (userInput.trim().toLowerCase().contains("done")) {
+                markTaskDone(taskList, userInput);
+            } else if (userInput.trim().toLowerCase().contains("todo")) {
+                listIndex = addToList(taskList, listIndex, userInput, "todo");
+            } else if (userInput.trim().toLowerCase().contains("deadline")) {
+                listIndex = addToList(taskList, listIndex, userInput, "deadline");
+            } else if (userInput.trim().toLowerCase().contains("event")) {
+                listIndex = addToList(taskList, listIndex, userInput, "event");
+            } else {
+                printErrorMessage();
+            }
+            userInput = sc.nextLine();
+        }
+    }
+
+    public static void markTaskDone(Task[] taskList, String userInput) {
+        Task taskDone = getCompletedTask(taskList, userInput);
+        taskDone.markAsDone();
+        printDivider();
+        System.out.println("\tVery good, sir. Excellent work accomplishing your task:");
+        System.out.print('\t');
+        System.out.println(taskDone);
+        printDivider();
+    }
+
+    public static Task getCompletedTask(Task[] taskList, String userInput) {
+        String replacedInput = userInput.trim().toLowerCase().replace("done", "");
+        int taskNumber = Integer.parseInt(replacedInput.trim().substring(0, 1));
+        return taskList[taskNumber-1];
+    }
+
+    public static int addToList(Task[] taskList, int listIndex, String userInput, String taskType) {
+        String replacedInput;
+        String description;
+        switch (taskType) {
+        case "todo":
+            replacedInput = userInput.replaceFirst("(?i)todo", "").trim();
+            description = replacedInput;
+            taskList[listIndex] = new ToDo(description);
+            break;
+        case "deadline":
+            replacedInput = userInput.replaceFirst("(?i)deadline", "").trim();
+            int byIndex = replacedInput.toLowerCase().indexOf("/by");
+            String by = replacedInput.substring(byIndex + 3);
+            description = replacedInput.substring(0, byIndex);
```

Comment:
you can consider using the split() method to separate the description and by strings, with " /by " as the separator, this way it'll save a few lines and we need not count the number of characters for substrings